Add docs for analyzers ASP0011-14 (#27109)
dotnet/AspNetCore.Docs: aspnetcore/diagnostics/asp0014.md

---
title: "ASP0014: Suggest using top level route registrations"
description: "Learn about analysis rule ASP0014: Suggest using top level route registrations"
author: safia
monikerRange: '>= aspnetcore-7.0'
ms.author: safia
ms.date: 09/27/2022
uid: diagnostics/asp0014
---
# ASP0014: Suggest using top level route registrations
| | Value |
|-|-|
| **Rule ID** |ASP0014|
| **Category** |Usage|
| **Fix is breaking or non-breaking** |Non-breaking|

## Cause

Routes can be registered directly at the top-level of a minimal API application.

## Rule description

> Routes can be registered directly at the top-level of a minimal API application and do not need to be nested within a `UseEndpoints` call.

Reviewer: ```suggestion
Routes can be registered directly at the top-level of a minimal API application and don't need to be nested within a `UseEndpoints` call.
```
